subsys/net_core_monitor/app_core.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/types.h>
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <nrfx_ipc.h>
+#include "common.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(net_core_monitor, CONFIG_NET_CORE_MONITOR_LOG_LEVEL);
+
+#define IPC_MEM_CNT_IDX  0
+#define IPC_MEM_REAS_IDX 1
+
+int ncm_net_status_check(uint32_t * const reset_reas)
+{
+	uint32_t gpmem;
+	uint16_t flags, cnt;
+	static uint16_t prv_cnt;
+
+	gpmem = nrfx_ipc_gpmem_get(IPC_MEM_CNT_IDX);
+	flags = (uint16_t)(gpmem >> FLAGS_POS);
+	cnt = (uint16_t)(gpmem);
+
+	if (flags & FLAGS_RESET) {
+		flags &= ~FLAGS_RESET;
+		gpmem = (gpmem & (~FLAGS_MSK)) | (flags << FLAGS_POS);
```

Comment:
Also this could be simplified:
```
	if (gpmem & (FLAGS_RESET << FLAGS_POS)) {
		gpmem &= ~(FLAGS_RESET << FLAGS_POS);
```
to reduce the risk of mistakes.